presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloPlugin.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 Bloomberg L.P.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.accumulo;
+
+import com.facebook.presto.metadata.FunctionFactory;
+import com.facebook.presto.spi.ConnectorFactory;
+import com.facebook.presto.spi.Plugin;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Main extension point for Accumulo connector for Presto. Provides services related to Accumulo to
+ * Presto, such as the connector itself and any defined UDFs.
+ */
+public class AccumuloPlugin
+        implements Plugin
+{
+    private TypeManager typeManager;
+    private Map<String, String> optionalConfig = ImmutableMap.of();
+
+    @Override
+    public synchronized void setOptionalConfig(Map<String, String> optionalConfig)
+    {
+        this.optionalConfig =
+                ImmutableMap.copyOf(requireNonNull(optionalConfig, "optionalConfig is null"));
+    }
+
+    @Inject
+    public synchronized void setTypeManager(TypeManager typeManager)
+    {
+        this.typeManager = typeManager;
+    }
+
+    public synchronized Map<String, String> getOptionalConfig()
+    {
+        return optionalConfig;
+    }
+
+    @Override
+    public synchronized <T> List<T> getServices(Class<T> type)
```

Comment:
Do you need this to be `synchronized`? Presto won't call it from multiple threads.